glusterd2/transaction/step.go

```diff
@@ -143,10 +145,18 @@ func runStepFuncOnNode(stepName string, ctx TxnCtx, node uuid.UUID, respCh chan<
 	respCh <- stepPeerResp{node, err}
 }
 
-func runStepFuncLocally(stepName string, ctx TxnCtx) error {
+func runStepFuncLocally(stepName string, ctx TxnCtx, origCtx ...context.Context) error {
```

Comment:
Do you expect a case in the future where you may have multiple origCtx's? If not, then it's better that this (and the other occurances above) be simple function rather than variadic functions (...origCtx -> origCtx).

Reply:
No, I don't expect multiple origCtx's. I have converted all the functions to simple functions.

My initial thought was to make the functions variadic so that this parameter need not be specified at all if tracing is not required for certain transactions. But looking closer, we needn't make them variadic. If the context parameter is nil, the transaction and associated step function(s) won't be traced.